### PR TITLE
Support sorting of chords and voices

### PR DESCRIPTION
**Pull request by shadanan**

### CODE REVIEW

File: lib/chord.js [hunk 1]
```diff
@@ -67,15 +71,43 @@ class Chord {
   transpose(amount) {
     return new Chord(transpose(this.root, amount), this.quality, transpose(this.bass, amount));
   }
+}
 
-  static isChord(str) {
-    return str.match(chordPattern) ? true : false;
+const compareNotes = (note1 = '', note2 = '') => NOTE_SEQUENCE.get(note1) - NOTE_SEQUENCE.get(note2);
```
Comment (vincelugli) [inline lib/chord.js]: Why do we need defaults in this? Is there ever a case where this should be called with on parameter?

File: lib/chord.js [hunk 2]
```diff
@@ -67,15 +71,43 @@ class Chord {
...
+  }
+  cmp = compareQualities(chord1.quality, chord2.quality);
+  if (cmp != 0) {
+    return cmp;
   }
+  return compareNotes(chord1.bass, chord2.bass);
+};
+
+function isChord(str) {
+  return str.match(CHORD_PATTERN) ? true : false;
```
Comment (vincelugli) [inline lib/chord.js]: This could be simplified to `return !!str.match(CHORD_PATTERN);`

**Review by vincelugli — APPROVED** [review]
👍

File: lib/chord.js (hunk 2 above)
Comment (shadanan) [reply lib/chord.js]: It's nice having someone who actually knows and likes Javascript.

File: lib/chord.js (hunk 1 above)
Comment (shadanan) [reply lib/chord.js]: Yes, you're right. The original logic here was that Chords have two notes, a root note and a bass note. The bass note is optional, but the way I wrote the Chord parser, the bass note will never be undefined, so removing the `''` does in fact work.

**Pull request merged**